pommerman/forward_model.py

```diff
@@ -159,7 +159,11 @@ def has_position_conflict(counter):
                     elif not agent.can_kick:
                         agent.stop()
                     else:
-                        next_positions[agent.agent_id] = next_position
+                        invalid_values = list(range(len(constants.Item)+1))[1:]
+                        if not utility.is_valid_direction(curr_board, next_position, action, invalid_values=invalid_values):
+                            agent.stop()
+                        else:
+                            next_positions[agent.agent_id] = next_position
```

Comment:
I see. So by the time we get to the else statement, we know that `next_position` is a bomb and the agent can kick. However, if the position _after_ `next_position` is not a passage, then we still don't want to move the agent. Good catch.

Let's do this slightly differently and say:
```
after_next_position = utility.get_next_position(next_position, action)
if not utility.position_on_board(curr_board, after_next_position) or not utility.position_on_passage(curr_board, after_next_position):
    agent.stop()
else:
    ...
```

Sg?

Reply:
Want to make that change on this PR?

Reply:
Sure! Made two slight changes to what you wrote, seems to work now.